src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -94,24 +94,25 @@ public static KNNCodecUtil.Pair getFloats(BinaryDocValues values) throws IOExcep
     }
 
     public static long calculateArraySize(int numVectors, int vectorLength, SerializationMode serializationMode) {
+        // For more details on array memory usage in Java, visit https://www.javamex.com/tutorials/memory/array_memory_usage.shtml
```

Comment:
Please mention that following is for 2-d array.
Also, it is true regardless 32bits and 64bits machine?

Reply:
Why would 32 bits and 64 handle it differently?

Reply:
I didn't see any mention of a difference for array size in my research

Reply:
The reference size or rounding could be different between 32 and 64.

Reply:
https://blog.ycrash.io/how-much-memory-does-your-java-object-occupy/

Reply:
Reference size would only matter if we had 3-D arrays, and we only have 2-D since it's an array of vectors.
For the rounding number, I'm a little confused because I thought no matter what in Java it always rounds up to a multiple of 8